pkgs/development/libraries/nvidia-thrust/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, config
+, fetchFromGitHub
+, stdenv
+, cmake
+, pkg-config
+, cudaSupport ? config.cudaSupport or false
+, cudaPackages
+, symlinkJoin
+, enableOpenmp ? true
+, enableTbb ? false
+, tbb
+, hostSystem ? "CPP"
+, deviceSystem ? let
+    devices = lib.optional cudaSupport "CUDA"
+      ++ lib.optional enableTbb "TBB"
+      ++ lib.optional enableOpenmp "OMP"
```

Comment:
Also -- more importantly -- I find this logic to be incredibly confusing. IIUC there are 4 possible values for `deviceSystem` (CPP, OMP, TBB, CUDA) and 3 possible values for `hostSystem` (CPP, OMP, TBB). What about just letting the user specify deviceSystem instead of using flags like `cudaSupport`, `enableTbb`, `enableOpenmp`?

Optionally adding a bunch of things to a list and then taking the its `head` is just kinda confusing IMHO. Having multiple, mutually-exclusive boolean flags violates the whole "make illegal states unrepresentable" principle.

Reply:
> illegal states irrepresentable

Agreed, now looking back the whole bally thing was a bit of an overkill

> list, `head`

Similarly top `optLevel` in faiss, the goal was to choose the best available level of acceleration.
I've now removed this logic for `deviceSystem`, but at the cost of somewhat obscuring `nvidia-thrust-intel`